### ISSUE TITLE
messaging tweaks

### ISSUE THREAD

**Issue by nightpool** [new-xkit/XKit#748]
ideas from the askbox

 - block messages altogether
 - de-gif gif headers
 - differentiate chat bubbles
 - timestamps!!

hard:

 - multiple chat windows
 - better message notifs

other ideas:

text has a "formatting" attribute. what does this do??
what other types of messages are there?

**Comment by rimbaum** [comment]
Another suggestion: A friend of mine noticed that if a blog isn't listed in search, there's no way to send a message to them from a sideblog. Clicking the "message" button on their blog page or from dash defaults to the main blog of your account, with no option to switch. If there's a workaround for this, it could be really helpful.

**Comment by ThePsionic** [comment]
@rimbaum What do you mean by "isn't listed in search"? As far as I know, if you type the exact URL into the search box you can find the blog no matter what.

**Comment by rimbaum** [comment]
@ThePsionic If a blog has "Allow this blog to show in search results" unchecked in the privacy options, they will not show up when sending a message unless they're following you. I just tested it with one of my old rp blogs (titaniumtwin) that had this option left unchecked, and it wouldn't allow me to send a thing from the messaging window.

However, if I go directly to the blog, it shows the message button. But clicking on the message button from the blog page will <em>only</em> let you send from your main blog. Same for the avatar menu option for messaging on your dash - main blog only if you select from there.

**Comment by ThePsionic** [comment]
@rimbaum Ah, I was not aware that that option had effect on the search function of messaging.

**Comment by Wolvan** [comment]
@rimbaum @ThePsionic the search feature for chats is based on the same limitations as the @-tagging tumblr has. I wouldn't even know how to start fixing this

**Comment by rimbaum** [comment]
@Wolvan Nah, the workaround is just to go to their blog and hit the message button. But again, it defaults to the main blog, leaving you with no way to send an unlisted blog a message from any of your sideblogs. Which is kind of what I was suggesting a feature for? The ability to choose a blog to send from when hitting the message button from either the user menu on dash, or the message button on a person's blog. That lets you sidestep the limitations in the chat search function.

**Comment by Wolvan** [comment]
@rimbaum ![image](https://cloud.githubusercontent.com/assets/1437010/11121462/46f03dce-8956-11e5-9653-974f276274ad.png)
Look what I found while digging through the IM's HTML

**Comment by rimbaum** [comment]
@Wolvan they sure don't make it easy to know stuff like that though, do they? :P

**Comment by Wolvan** [comment]
Probably a WIP, but so far it works well enough I suppose. Easy tweak to restore it, gonna add that for the next version of MT

**Comment by nightpool** [comment]
@wolvan I don't know if that does what you think it does... it looks like
it would change participants in the same conversation, rather then creating
a different conversation.

**Comment by Wolvan** [comment]
@nightpool it seems to create an entirely new conversation, try it out yourself. On the ``conversation-compose`` remove the ``hide-blog-selector`` class and remove the ``display:none`` style of ``conversation-compose-blog-selector`` (child of ``conversation-compose``)
